packages/remix-server-runtime/server.ts

```diff
@@ -406,60 +410,79 @@ async function singleFetchAction(
       signal: request.signal,
       ...(request.body ? { duplex: "half" } : undefined),
     });
-    let result = await staticHandler.queryRoute(handlerRequest, {
+
+    let responseStubs = getResponseStubs();
+    let result = await staticHandler.query(handlerRequest, {
       requestContext: loadContext,
+      skipLoaderErrorBubbling: true,
+      skipLoaders: true,
+      unstable_dataStrategy: getSingleFetchDataStrategy(responseStubs),
```

Comment:
We pass a factory-created `dataStrategy` to `query` now so we can have a set of `responseStub`'s scoped to this request